### ISSUE TITLE
Shutdown/Reboot breaks ability to boot from cloned disk

### ISSUE THREAD

**Issue by cRoCx** [rescuezilla/rescuezilla#337]
# Successful scenario
Cloning all partitions internal NVMe SSD to external USB HDD, overwriting all partitions and choosing not to shutdown or reboot after successful completion --> After manual restart I can boot my cloned Windows from the external USB HDD.

# Failing scenario (bug description)
Same as above, but instead choosing to shutdown or reboot after successful completion --> I can not boot my cloned Windows from the external USB HDD anymore. I can access all files, mount all partitions, but I cannot boot from the cloned disk. When I manually boot a Windows recovery environment and then try to boot from the HDD, it just bluescreens. Bitlocker is disabled.

# Notes
I cannot figure out what's causing this, since both scenarios throw no errors in Clonezilla, but something went wrong after introducing this feature in https://github.com/rescuezilla/rescuezilla/pull/205.

**Comment by shasheene** [comment]
Thanks for reporting the bug in Rescuezilla Weekly Rolling Release 2022-06-26.

That sounds strange. I'll investigate.

**Comment by shasheene** [comment]
I wasn't able to reproduce this issue on the 'Restore' mode for Rescuezilla Weekly Rolling Release 2022-06-26 on both MBR and GPT disks.

I ran out of time to test the 'Clone' mode with this, before releasing ["Weekly" Rolling Release (2022-07-24)](https://github.com/rescuezilla/rescuezilla/releases/tag/rolling-release.2022-07-24) that updates the base operating system.

I added a disclaimer to both weekly rolling releases and a message encouraging users to stick with the stable versions unless they're comfortable being beta testers.

I'll try to reproduce the 'Clone' mode before Rescuezilla v2.4 releases next week.

**Comment by shasheene** [comment]
I've tested the 'Clone' mode with a variety of MBR and GPT Windows disks with Rescuezilla v2.4, and **was still unable to reproduce the error in my virtual machine testing environment**.

Because I can't yet reproduce this, and I have received no other reports, I am release Rescuezilla v2.4 as-is with this situation listed as a possible error.

Hopefully anybody else who experiences this can provide further context.

**Comment by cRoCx** [comment]
I haven't tried it since. Will do again in the next few days.

**Comment by lxwulf** [comment]
I think it could be because you can't install (or boot) windows on an external HDD...

I had some experience with it and ended on WindowsToGo installation from Rufus application. Maybe you could do a Test installation with WindowsToGo, clone it and look if its start when yes you know where the problem is...

This was just my first thoughts about it. 🤔

**Comment by shasheene** [comment]
An anonymous user [wrote](https://sourceforge.net/p/rescuezilla/discussion/support/thread/19f25ba22b/?limit=25#257e) the following:
> Bug: <this ticket>
> The reason is that rescuezilla breaks the drives. The reason is that users use GUIs. It does not happen to people using Konsole/Terminal. Writing to disks may not be finished and some GUI software or because of any other reason some software keep the drive busy. You will break the file system and/or drive when you unplug/shutdown/reboot.
> 
> I personally don't use GUIs for copying. I use rsync from terminal and then (!)
$ udisksctl unmount -b /dev/sdX1
$ udisksctl power-off -b /dev/sdb
> 
> YOU MUST first unmount and power off the drive that you were writing to. This ensures no operations are done to the drive. External drives DO NOT have power-off button. You will make NTFS dirty and broken. If you try to repair it in Windows any distro will fail to read/write/see it and bydoing chkdsk etc you may loose data. Similar errors appear on other file systems (ext4, xfs, etc). The fact that they (may) work after hard reset is their resilience but it does not justify of Rescuzilla doing HARD RESET on them!
> 
> Either add two of the above commands to shutdown and reboot to be done automatically before it really reboots/shutdowns or remove this option completely as it breaks the sdrives.
> 
> Thank you for reading.
> Great software anyway!

**Comment by shasheene** [comment]
The above anonymous user's suggestion is interesting.

Rescuezilla does unmount partitions of course, but after scheduling a shutdown it doesn't do any do any special commands (like commanding the drive to power-off). As best I can tell, those commands shouldn't be required, but it's possible I'm missing a service that makes sending the power-off command required.

The user mentions the `udisks2` service, which is responsible for managing disks on the system (for tools like the file manager). There is a risk that Rescuezilla (and GParted) could conflict with it, which is why Rescuezilla disables it while it's running: the file manager's list of disks disappears when Rescuezilla or GParted is opened, and re-appears when it's closed.

It's possible that sending a shutdown request means the system has a shutdown without fully flushing the drive's volatile cache to disk, but I doubt that's the case too. Unless I'm mistaken Linux should prevent the shutdown until all the caches are flushed to disk.

I'll keep trying to reproduce this using the equipment I have. @cRoCx you saw this issue on a Windows EFI/GPT environment, not a legacy MBR/BIOS right? And it was just plain Windows? Not a dualboot with Linux involving the GRUB bootloader too?

**Comment by cRoCx** [comment]
Yes, @shasheene, the source SSD was GPT with a Windows EFI. No dualboot. See the screenshot below. Disk 1 is the internal SSD. Disk 2 is the cloned result.  ![2022-08-10 Rescuezilla Clone with Shutdown](https://user-images.githubusercontent.com/10899060/184016053-bfa52a5d-f676-455d-9271-2a68ff67f6b8.png) You can also see, that I use Bitlocker now, but at the time of reporting the issue, I didn't use it.

Unfortunately I cannot test it properly again. After cloning and rebooting, whenever I select to boot from USB it boots the EFI from USB, but then automatically switches to my internal SSD Windows partition for the rest of the boot process, without showing any errors. So half of the boot process runs from USB and the other half from the SSD. Weird Windows behaviour. I cannot force it to not use my internal SSD. This happens in both scenarios (having Clonezilla auto shutdown and not auto shutting down after cloning).

I also don't want to remove the internal SSD just for this re-test, since I almost broke the back cover the last time I removed it. :-(

**Comment by shasheene** [comment]
Oh I think I understand exactly what happened: after you successfully completed your clone, your destination disk now contains identical "UUID" label.

When your computer boots, your disk's bootloader uses labels like the "UUID" to identify the disk uniquely. With duplicate disks, this can lead to the exact kind of conflict your describe.

It's highly recommended you disconnect one of your disks, because operating systems often don't expect this conflict to happen, and it can easily cause bigger problems (like inconsistently booting the wrong disk).

**Comment by cRoCx** [comment]
Oh yes, that explaines my test scenario from this week. Thank you very much for this!

But it wouldn't explain my original scenario. Or does it?
Because there I had an SSD cloned to USB HDD, then removed the SSD from the laptop, installed another SSD with larger capacity and then tried to clone back from USB HDD to the new SSD.

**Comment by mhmatthewhugley** [comment]
I cloned from a bigger HDD to SSD by manually shrinking the partition and increasing the SSDs partition after cloning. I selected the shutdown option for when cloning completed and everything worked as intended.

**Comment by 0xC4aE1e5** [comment]
This could be because the successful scenario had enough time to sync. But Windows doesn't like it very much when the system drive letter changes.
